### PR TITLE
RDoc-1906_RollingIndexDeployment

### PR DESCRIPTION
**Pull request by reebhub**

### CODE REVIEW

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+---
+
+### The Rolling Procedure
+
+Nodes are assigned with the indexing of each database in a linear order, one node at a time.  
+
+1. The cluster assigns indexing to one of its nodes.  
+2. When the assigned node finishes indexing, it send the cluster leader a confirmation 
+   that indexing is done.
```
Comment (karmeli87) [inline Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:58]:
```suggestion
2. When the assigned node finishes indexing, it send a cluster-wide confirmation command
   that indexing is done.  
```

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+---
+
+### The Rolling Procedure
+
+Nodes are assigned with the indexing of each database in a linear order, one node at a time.  
+
+1. The cluster assigns indexing to one of its nodes.  
+2. When the assigned node finishes indexing, it send the cluster leader a confirmation 
+   that indexing is done.  
+3. The cluster leader assigns indexing to the next node.
```
Comment (karmeli87) [inline Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:59]:
```suggestion
3. The cluster assigns indexing to the next node.  
```

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+1. The cluster assigns indexing to one of its nodes.  
+2. When the assigned node finishes indexing, it send the cluster leader a confirmation 
+   that indexing is done.  
+3. The cluster leader assigns indexing to the next node.  
+   {WARNING: }
+   If the delivery of an **indexing completion confirmation** fails, e.g. because the 
+   sending (indexing) or receiving (cluster leader) node has been forcefully disconnected 
+   from the cluster, **indexing will pend** for all nodes until the cluster recovers or 
+   indexing is initiated manually.
```
Comment (karmeli87) [inline Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:64]:
```suggestion
   If the delivery of an **indexing completion confirmation** fails,  **indexing will pend** for all nodes until the confirmation succeed or 
   indexing is initiated manually.  
```

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+   {WARNING/}
+
+---
+
+### Deployment Scope and Course
+
+* **Rolling Scope**  
+  The cluster maintains an **independent indexing deployment course** for each databases.  
+* **Rolling Order**  
+  Rolling order is determined by the cluster (node `A` **not** necessarily always being the first).
```
Comment (karmeli87) [inline Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:74]: The initial order is the reverse order of the Member nodes of the database group concatenate with Rehabs and lastly with the Promotables

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+The implication of this design is that different nodes **can** index different databases concurrently.  
+
+* E.g. -  
+   * The indexing deployment course the cluster chose for the **"Integration"** database, is:  
+     indexing by node `B`, then by node `C`, and finally by node `A`.  
+   * The indexing deployment course the cluster chose for the **"Production"** database, is:  
+     indexing by node `C`, then by node `A`, and finally by node `B`.  
+   * Assuming that the cluster has started indexing the two databases at the same time,  
+     node `B` will be indexing "Integration" while node `C` indexes "Production".
```
Comment (karmeli87) [inline Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:84]: If find it hard to follow... In general, each _index_ can have it's own order of deployment

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:58]: done

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown (hunk 3 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:64]: If the delivery of an **indexing completion confirmation** fails, **indexing will pend** for all nodes until the confirmation succeeds or indexing is initiated manually.  
Confirmation delivery may fail, for example, due to forceful disconnection of the indexing node or cluster leader node during indexation.

Comment (karmeli87) [reply Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:64]: Actually to say that "indexing will pend for all nodes" is not accurate.
What will happened is that the running index will complete its job, but will be unable to notify about that. 
So no other index can start running

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown (hunk 4 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:74]: * **Rolling Order**  
  The rolling order is determined by the cluster.  
  Indexing is deployed to nodes in the reverse order of their membership in the database group, giving a lower priority to nodes in [Rehab or Promotable](../server/clustering/distribution/distributed-database#database-topology) state.

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown (hunk 5 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:84]: removed. the section is now "Deployment Concurrency and Order", and this part is:

* **Deployment Concurrency**  
   * A node is assigned to index a database, only if no other node currently indexes this database.  
   * Multiple nodes **can** be assigned to concurrently index **different databases**.  
     E.g., node `A` can index the "Integration" database, while node `B` indexes the "Production" database.

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown (hunk 2 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown:59]: done

File: Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown (hunk 3 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/indexes/rolling-index-deployment.markdown]: done:

If the delivery of an **indexing completion confirmation** fails when the current node finishes indexing, no other node will be able to start indexing until the confirmation succeeds or indexing is initiated manually.

**Review by karmeli87 — APPROVED** [review]
